From 89cc9474996a26131c82838f4c0671f1ef7bab03 Mon Sep 17 00:00:00 2001
From: yvanliang <yvan6liang@outlook.com>
Date: Fri, 9 Oct 2026 15:40:30 +0800
Subject: [PATCH 06/60] GSI Android 17 compatibility:
 0007-Re-order-services-so-that-it-works-even-without-qtag.patch

Change-Id: Iad9466ec7a6dfac5f5ae21dde56e81ef206d22a5
---
 .../android/server/net/NetworkPolicyManagerService.java   | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/services/core/java/com/android/server/net/NetworkPolicyManagerService.java b/services/core/java/com/android/server/net/NetworkPolicyManagerService.java
index 96a9e9579..fb6316e37 100644
--- a/services/core/java/com/android/server/net/NetworkPolicyManagerService.java
+++ b/services/core/java/com/android/server/net/NetworkPolicyManagerService.java
@@ -1066,6 +1066,10 @@ public class NetworkPolicyManagerService extends INetworkPolicyManager.Stub {
         Trace.traceBegin(Trace.TRACE_TAG_NETWORK, "systemReady");
         final int oldPriority = Process.getThreadPriority(Process.myTid());
         try {
+            mUsageStats = LocalServices.getService(UsageStatsManagerInternal.class);
+            mAppStandby = LocalServices.getService(AppStandbyInternal.class);
+            mActivityManagerInternal = LocalServices.getService(ActivityManagerInternal.class);
+
             // Boost thread's priority during system server init
             Process.setThreadPriority(Process.THREAD_PRIORITY_FOREGROUND);
             if (!isBandwidthControlEnabled()) {
@@ -1073,10 +1077,6 @@ public class NetworkPolicyManagerService extends INetworkPolicyManager.Stub {
                 return;
             }
 
-            mUsageStats = LocalServices.getService(UsageStatsManagerInternal.class);
-            mAppStandby = LocalServices.getService(AppStandbyInternal.class);
-            mActivityManagerInternal = LocalServices.getService(ActivityManagerInternal.class);
-
             mUseMeteredFirewallChains = Flags.useMeteredFirewallChains();
             mUseDifferentDelaysForBackgroundChain = Flags.useDifferentDelaysForBackgroundChain();
             mNeverApplyRulesToCoreUids = Flags.neverApplyRulesToCoreUids();
